packages/database/src/migrations/20200928065206-CreateLaosSchoolsTeachingAndLearningBarGraphDashboardReport-modifies-data.js

```diff
@@ -0,0 +1,208 @@
+'use strict';
+
+import { insertObject, arrayToDbString } from '../utilities/migration';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function(options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const PRE_SCHOOL = 'Pre-School';
+const PRIMARY_SCHOOL = 'Primary';
+const SECONDARY_SCHOOL = 'Secondary';
+const DASHBOARD_GROUP_CODES = [
+  'LA_Laos_Schools_Country_Laos_Schools_User',
+  'LA_Laos_Schools_Province_Laos_Schools_User',
+  'LA_Laos_Schools_District_Laos_Schools_User',
+];
+const REPORT_ID = 'Laos_Schools_Teaching_Learning_Bar_Graph';
+
+const BASE_SUB_BUILDER_CONFIG = {
+  dataClasses: {
+    'Remedial support provided to students by the school': {
+      numerator: {
+        valueOfInterest: 'Yes',
+        dataValues: ['SchFF011'],
+      },
+      denominator: {
+        valueOfInterest: '*',
+        dataValues: ['SchFF011'],
+      },
+      sortOrder: 0,
+    },
+    'Support implementing catch-up/remedial teaching programmes received': {
+      numerator: {
+        valueOfInterest: 'Yes',
+        dataValues: ['SchCVD020'],
+      },
+      denominator: {
+        valueOfInterest: '*',
+        dataValues: ['SchCVD020'],
+      },
+      sortOrder: 1,
+    },
+    'Additional learning and reading materials received since June 2020': {
+      numerator: {
+        valueOfInterest: 'Yes',
+        dataValues: ['SchCVD004b'],
+      },
+      denominator: {
+        valueOfInterest: '*',
+        dataValues: ['SchCVD004b'],
+      },
+      sortOrder: 2,
+    },
+    'Teachers follow the MoES education programmes at home': {
+      numerator: {
+        valueOfInterest: 'Yes',
+        dataValues: ['SchCVD016'],
+      },
+      denominator: {
+        valueOfInterest: '*',
+        dataValues: ['SchCVD016'],
+      },
+      sortOrder: 3,
+    },
+    'Students follow the MoES education programmes at home': {
+      numerator: {
+        valueOfInterest: 'Yes',
+        dataValues: ['SchCVD017'],
+      },
+      denominator: {
+        valueOfInterest: '*',
+        dataValues: ['SchCVD017'],
+      },
+      sortOrder: 4,
+    },
+    'MoES education programmes on TV/radio/online considered useful for supporting continuity of student learning when schools were closed': {
```

Comment:
Does this require custom logic to handle such long names? In the screenshot provided in the card, the names are displayed diagonally and cut off so that the start of each can be seen. Could you please check that this is not a requirement?

Reply:
Just an update, Erin and Megan are fine with it for now: https://github.com/beyondessential/tupaia-backlog/issues/1063#issuecomment-700372164